topostats/tracing/skeletonize.py

```diff
@@ -2,78 +2,641 @@
+        """
+        return 2 <= self.p2 + self.p3 + self.p4 + self.p5 + self.p6 + self.p7 + self.p8 + self.p9 <= 6
+
+    def _binaryThinCheck_b_returncount(self):
+        """Assess local area connectivity?"""
+        count = sum(
+            [
+                [self.p2, self.p3] == [0, 1],
+                [self.p3, self.p4] == [0, 1],
+                [self.p4, self.p5] == [0, 1],
+                [self.p5, self.p6] == [0, 1],
+                [self.p6, self.p7] == [0, 1],
+                [self.p7, self.p8] == [0, 1],
+                [self.p8, self.p9] == [0, 1],
+                [self.p9, self.p2] == [0, 1],
+            ]
+        )
+
+        return count
+
+    def _binaryThinCheck_c(self):
+        """Check if p2, p4 or p6 is 0 - seems very specific
+
+        Returns
+        -------
+        bool
+            if p2, p4 or p6 is 0.
+        """
+        return self.p2 * self.p4 * self.p6 == 0
+
+    def _binaryThinCheck_d(self):
+        """Check if p4, p6 or p8 is 0 - seems very specific
 
-def _get_skeletonize(method: str = "zhang") -> Callable:
-    """Creator component which determines which skeletonize method to use.
+        Returns
+        -------
+        bool
+            if p4, p6 or p8 is 0.
+        """
+        return self.p4 * self.p6 * self.p8 == 0
 
-    Parameters
-    ----------
-    method: str
-        Method to use for skeletonizing, methods are 'zhang' (default), 'lee', 'medial_axis', and 'thin'.
+    def _binaryThinCheck_csharp(self):
+        """Check if p2, p4 or p8 is 0 - seems very specific
 
-    Returns
-    -------
-    Callable
-        Returns the function appropriate for the required skeletonizing method.
+        Returns
+        -------
+        bool
+            if p2, p4 or p8 is 0.
+        """
+        return self.p2 * self.p4 * self.p8 == 0
+
+    def _binaryThinCheck_dsharp(self):
+        """Check if p2, p6 or p8 is 0 - seems very specific
+
+        Returns
+        -------
+        bool
+            if p2, p6 or p8 is 0.
+        """
+        return self.p2 * self.p6 * self.p8 == 0
+
+    def finalSkeletonisationIteration(self):
+        """A final skeletonisation iteration that removes "hanging" pixels.
+        Examples of such pixels are:
+                    [0, 0, 0]               [0, 1, 0]            [0, 0, 0]
+                    [0, 1, 1]               [0, 1, 1]            [0, 1, 1]
+            case 1: [0, 1, 0]   or  case 2: [0, 1, 0] or case 3: [1, 1, 0]
+
+        This is useful for the future functions that rely on local pixel environment
+        to make assessments about the overall shape/structure of traces"""
+
+        remaining_coordinates = np.argwhere(self.mask).tolist()
+
+        for x, y in remaining_coordinates:
+            self.p7, self.p8, self.p9, self.p6, self.p2, self.p5, self.p4, self.p3 = self.getLocalPixelsBinary(
+                self.mask, x, y
+            )
+
+            # Checks for case 1 pixels
```

Comment:
I suspect this is copied verbatim but we could perhaps describe what `case 1` and `case 2` are from the Zhang paper.

Reply:
These are described in `final_skeletonisation_iteration`